dev/gen_version.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# This file should be called from the top-level directory of Coq!
+export LC_ALL=C
+COMPILE_DATE=`date`
+
+# default coq version
+COQ_VERSION=8.8+alpha
+
+echo "type version = { describe : string; branch : string; hash : string } ;;"
+
+if [ -x `which git` ] && [ -d .git ]
+then
+    GIT_DESCRIBE=`git describe --tags --first-parent --dirty | sed 's/^V//'`
+    GIT_BRANCH=`git branch -a | sed -ne '/^\* /s/^\* \(.*\)/\1/p'`
+    GIT_HASH=`git rev-parse HEAD`
+    echo "let version = { describe = \"${GIT_DESCRIBE}\"; branch = \"${GIT_BRANCH}\"; hash = \"${GIT_HASH}\" } ;;"
+else
+    echo "let version = { describe = \"${COQ_VERSION}\"; branch = \"unknown\"; hash = \"\" } ;;"
```

Comment:
That's not a criticism against @ejgallego's commit because I think it was already the case before (with just @maximedenes' changes) but the result will be `coqtop -v` showing:

The Coq Proof Assistant, version 8.8.0 (unknown)

when it used to show:

The Coq Proof Assistant, version 8.6 (June 2017)

I would need to look at @maximedenes' commits again to understand how it moved from showing a date to a branch but I feel that for released versions we distribute, it doesn't make much sense to show the branch, especially if it is "unknown".
Could the branch be turned into an option to avoid showing "(unknown)" or would that be too much work?

Reply:
With @ejgallego 's commits, I think it will show `The Coq Proof Assistant, version 8.8.0 (v8.8)`, assuming we distribute `version/coqversion.ml`.  I think we have two options here:
1. Commit to distributing `coqversion.ml`, always display the branch
2. Don't distribute `coqversion.ml`, generate it with an option, display `Option.default Coq_config.date Coqversion.version.branch` in places where we currently display the branch